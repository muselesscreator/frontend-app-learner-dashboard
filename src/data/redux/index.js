import { combineReducers } from 'redux';

import { StrictDict } from 'utils';

import * as app from './app';
import * as requests from './requests';

export { default as thunkActions } from './thunkActions';

const modules = {
  app,
  requests,
};

const moduleProps = (propName) => Object.keys(modules).reduce(
  (obj, moduleKey) => ({ ...obj, [moduleKey]: modules[moduleKey][propName] }),
  {},
);

const rootReducer = combineReducers(moduleProps('reducer'));

const actions = StrictDict(moduleProps('actions'));

const selectors = StrictDict(moduleProps('selectors'));

export { actions, selectors };

export default rootReducer;
