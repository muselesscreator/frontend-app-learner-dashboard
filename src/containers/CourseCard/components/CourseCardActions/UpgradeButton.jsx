import React from 'react';
import PropTypes from 'prop-types';

import { Button } from '@edx/paragon';
import { Locked } from '@edx/paragon/icons';
import { useIntl } from '@edx/frontend-platform/i18n';

import { hooks } from 'data/redux';
import messages from './messages';

export const UpgradeButton = ({ cardId }) => {
  const { upgradeUrl } = hooks.useCardCourseRunData(cardId);
  const { canUpgrade } = hooks.useCardEnrollmentData(cardId);
  const { isMasquerading } = hooks.useMasqueradeData();
  const { formatMessage } = useIntl();
  if (!upgradeUrl) {
    return null;
  }
  return (
    <Button
      iconBefore={Locked}
      variant="outline-primary"
      disabled={isMasquerading || !canUpgrade}
      as="a"
      href={upgradeUrl}
    >
      {formatMessage(messages.upgrade)}
    </Button>
  );
};
UpgradeButton.propTypes = {
  cardId: PropTypes.string.isRequired,
};
export default UpgradeButton;
