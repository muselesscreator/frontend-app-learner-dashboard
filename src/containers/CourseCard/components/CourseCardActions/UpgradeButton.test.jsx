import { shallow } from 'enzyme';

import { htmlProps } from 'data/constants/htmlKeys';
import { hooks } from 'data/redux';
import UpgradeButton from './UpgradeButton';

jest.mock('data/redux', () => ({
  hooks: {
    useMasqueradeData: jest.fn(() => ({ isMasquerading: false })),
    useCardCourseRunData: jest.fn(),
    useCardEnrollmentData: jest.fn(() => ({ canUpgrade: true })),
  },
}));

describe('UpgradeButton', () => {
  const props = {
    cardId: 'cardId',
  };
  const upgradeUrl = 'upgradeUrl';
  hooks.useCardCourseRunData.mockReturnValue({ upgradeUrl });
  describe('snapshot', () => {
    test('can upgrade', () => {
      const wrapper = shallow(<UpgradeButton {...props} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.prop(htmlProps.disabled)).toEqual(false);
    });
    test('no upgrade url', () => {
      hooks.useCardCourseRunData.mockReturnValueOnce({});
      const wrapper = shallow(<UpgradeButton {...props} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.isEmptyRender()).toEqual(true);
    });
    test('cannot upgrade', () => {
      hooks.useCardEnrollmentData.mockReturnValueOnce({ canUpgrade: false });
      const wrapper = shallow(<UpgradeButton {...props} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.prop(htmlProps.disabled)).toEqual(true);
    });
    test('masquerading', () => {
      hooks.useMasqueradeData.mockReturnValueOnce({ isMasquerading: true });
      const wrapper = shallow(<UpgradeButton {...props} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.prop(htmlProps.disabled)).toEqual(true);
    });
  });
});
