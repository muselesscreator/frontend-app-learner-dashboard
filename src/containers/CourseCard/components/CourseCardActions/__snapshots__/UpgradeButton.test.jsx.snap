// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpgradeButton snapshot can upgrade 1`] = `
<Button
  as="a"
  disabled={false}
  href="upgradeUrl"
  iconBefore={[MockFunction icons.Locked]}
  variant="outline-primary"
>
  Upgrade
</Button>
`;

exports[`UpgradeButton snapshot cannot upgrade 1`] = `
<Button
  as="a"
  disabled={true}
  href="upgradeUrl"
  iconBefore={[MockFunction icons.Locked]}
  variant="outline-primary"
>
  Upgrade
</Button>
`;

exports[`UpgradeButton snapshot masquerading 1`] = `
<Button
  as="a"
  disabled={true}
  href="upgradeUrl"
  iconBefore={[MockFunction icons.Locked]}
  variant="outline-primary"
>
  Upgrade
</Button>
`;

exports[`UpgradeButton snapshot no upgrade url 1`] = `""`;
